gadget/install/install.go

```diff
@@ -549,13 +549,20 @@ func MountVolumes(onVolumes map[string]*gadget.Volume, encSetupData *EncryptionS
 			if part.Filesystem == "" {
 				continue
 			}
-			mntPt := filepath.Join(boot.InitramfsRunMntDir, part.Name)
-			if part.Role == gadget.SystemSeedNull {
-				// Mount point is the same as partitions with system-seed role
+			var mntPt string
+			switch part.Role {
+			case gadget.SystemSeed, gadget.SystemSeedNull:
 				mntPt = boot.InitramfsUbuntuSeedDir
-
-				seedNullMntDir = mntPt
-				numSeedNull++
+				seedMntDir = mntPt
+				numSeedPart++
+			case gadget.SystemBoot:
+				mntPt = boot.InitramfsUbuntuBootDir
+			case gadget.SystemSave:
+				mntPt = boot.InitramfsUbuntuSaveDir
+			case gadget.SystemData:
+				mntPt = boot.InstallUbuntuDataDir
+			default:
+				mntPt = filepath.Join(boot.InitramfsRunMntDir, part.Name)
```

Comment:
shouldn't the fallback use Label as Run anyway?  or should we move Run to use the same code as here anyway?